src/main/java/com/rapid7/net/AsyncLogger.java

```diff
@@ -508,38 +511,33 @@ void dbg(String msg) {
         /**
          * Opens connection to InsightOps.
          *
-         * @throws IOException
+         * @throws IOException Thrown if we fail to connect
          */
         void openConnection() throws IOException {
-            try {
-                if (this.iopsClient == null) {
-                    this.iopsClient = new InsightOpsClient(httpPut, ssl, useDataHub, dataHubAddr, dataHubPort, region);
-                }
-                this.iopsClient.connect();
-
-                if (httpPut) {
-                    final String f = "PUT /%s/hosts/%s/?realtime=1 HTTP/1.1\r\n\r\n";
-                    final String header = String.format(f, key, location);
-                    byte[] temp = header.getBytes(ASCII);
-                    this.iopsClient.write(temp, 0, temp.length);
-                }
-
-            } catch (Exception e) {
+            if (this.iopsClient == null) {
+                this.iopsClient = new InsightOpsClient(httpPut, ssl, useDataHub, dataHubAddr, dataHubPort, region);
+            }
+            this.iopsClient.connect();
 
+            if (httpPut) {
+                final String f = "PUT /%s/hosts/%s/?realtime=1 HTTP/1.1\r\n\r\n";
+                final String header = String.format(f, key, location);
+                byte[] temp = header.getBytes(ASCII);
+                this.iopsClient.write(temp, 0, temp.length);
             }
         }
 
         /**
          * Tries to opens connection to InsightOps until it succeeds.
          *
-         * @throws InterruptedException
+         * @throws InterruptedException Thrown when interrupted while sleeping >:(
```

Comment:
You throw these at your alarm clock every morning yeah? :D

Reply:
all a thread wants is to sleep peacefully